reconcile/utils/openshift_resource.py

```diff
@@ -431,6 +433,15 @@ def calculate_sha256sum(body):
         return m.hexdigest()
 
 
+def checkExpirationDate(role):
+    exp_date = datetime.datetime \
+        .strptime(role['expirationDate'], '%Y-%m-%d').date()
```

Comment:
as far as i have seen the schema change, expirationDate is an optional field so you need to check if it is really in the role dict. otherwise you get a KeyError.

you should also make sure to handle the exception from an expirationDate that is not following the defined format.

can you also write tests for that?

Reply:
If this comes from the graphql query, the field will always exist but may be `None`. You are more likely to get a `TypeError`. :(
As @geoberle said, let's have a test for this.

Reply:
i always forgett about graphql returning None for those kind of things :(
thanks for the reminder!

Reply:
I wrote some tests for this but I don't believe I wrote is really checking what we want? Can anyone confirm if my latest commits are showing this?

Reply:
i saw that you introduced some methods to check the validity of the expiration date but i can only see them used from within the tests